include/reference_wrapper_helper.h

```diff
@@ -490,6 +490,17 @@ namespace quda
     vector() = default;
     vector(uint64_t size, const T &value = {}) : std::vector<T>(size, value) { }
 
+    /**
+       Constructor from pair of iterators
+       @param[in] first Begin iterator
+       @param[in] last End iterator
+     */
+    template <class U, std::enable_if_t<is_iterator_v<U>> * = nullptr> vector(U first, U last)
+    {
+      std::vector<T>::reserve(last - first);
+      for (auto it = first; it != last; it++) std::vector<T>::push_back(*it);
```

Comment:
`std::vector` supports using a begin and end iterator in the constructor (number 5 here: https://en.cppreference.com/w/cpp/container/vector/vector ); is there a reason that doesn't work here?

Reply:
44d8a2a1b17bda0bc4fac1b071d50ac225359d3c